torq.sh

```diff
@@ -0,0 +1,263 @@
+    checkinput $@                                                                                   # check input process names
+  fi
+ }
+
+flag() {
+  count=0
+  for i in ${BASH_ARGV[*]};
+  do
+    count=$(($count+1))
+    if [[ $i == "-$1" ]]; then                                                                      # find flag argument in command line
+      N=$(($count-2))                                                                               # find index of flag arugment 
+    fi
+  done
+ }
+
+flagextras() {
+  flag $@
+  z=$N
+  while [ $z -ge 0 ]
+  do
+    EXTRAS+="${BASH_ARGV[$z]} "                                                                     # get all parameters following extras flag
+    z=$[$z-1]
+  done
+ }
+
+flagcsv() {
+  flag $@
+  CSVPATH="${BASH_ARGV[$N]}"                                                                        # assign specifed csv file
+ }
+
+getextras() {
+  if [[ `echo ${BASH_ARGV[*]} | grep -e extras` ]]; then                                            
+    eval flagextras "extras";
+    length=$(($#-$N-2));
+    array=${@:1:$length};                                                                           # arguments without extras flag
+  else
+    array=$@;
+  fi
+ }
+
+getcsv() {
+  if [[ `echo ${BASH_ARGV[*]} | grep -e csv` ]]; then
+    eval flagcsv "csv";
+    length=$(($#-2));
+    array=${@:1:$length};                                                                           # arguments without csv flag
+    getprocs $array;
+  else
+    CSVPATH=${DEFAULTCSV};                                                                          # set csv file to default
+    getprocs $array;
+  fi
+ }
+
+getextrascsv() {
+  eval flagextras "extras";
+  eval flagcsv "csv";
+  length=$(($#-$N-2));
+  array=${@:1:$length};                                                                             # arguments without extras and csv flag
+  getprocs $array;
+ }
+
+checkextrascsv() {
+  if [[ `echo ${BASH_ARGV[*]} | grep -e extras` ]] && [[ `echo ${BASH_ARGV[*]} | grep -e csv` ]]; then
+    getextrascsv $@;
+  else
+    getextras $@;
+    getcsv $@;
+  fi
+ }
+
+allcsv() {
+  if [[ `echo ${BASH_ARGV[*]} | grep -e csv` ]]; then
+    eval flagcsv "csv";
+  else
+    CSVPATH=${DEFAULTCSV};                                                                            
+  fi
+ }
+
+startprocs() {
+  for p in $PROCS;
+  do
+    start $p;
+  done
+ }
+
+stopprocs() {
+  for p in $PROCS;
+  do
+    stop $p;
+  done
+ }
+
+usage() {
+  printf -- "Arguments:\n"
+  printf -- "-procs       	                        to list all processes\n"
+  printf -- "-summary             	                to view summary table\n"
+  printf -- "<processname> -debug			to debug process\n"
+  printf -- "start all -print                   	to view all default startup lines\n"
+  printf -- "start <processname(s)> -print		to view default startup lines\n"
+  printf -- "start all		           	to start all processes\n"
```

Comment:
looks like incorrect alignment on this line

Reply:
If I correctly align this in the script it doesn't align when the usage statement prints to screen.

Reply:
Hmm, seems strange, are tabs being used, or spaces? (it should be spaces) 

If everything lines up when it's run, that's fine then

Reply:
There's tabs in there, you can see if you highlight over the whitespace.